src/test/java/org/apache/commons/collections4/iterators/PairedIteratorTest.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.collections4.iterators;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Random;
+import java.util.UUID;
+import org.apache.commons.collections4.IteratorUtils;
+import org.apache.commons.collections4.iterators.PairedIterator.PairedItem;
+
+
+/** Unit test suite for {@link ZippingIterator}. */
+public final class PairedIteratorTest
+    extends AbstractIteratorTest<PairedItem<String, Integer>> {
+
+    //------------------------------------------------------------ Conventional
+
+    public PairedIteratorTest(String testName) {
+        super(testName);
+    }
+
+    //------------------------------------------------------------ Lifecycle
+
+    private ArrayList<String> smallStringsList = null;
+    private ArrayList<String> largeStringsList = null;
+    private ArrayList<Integer> smallIntsList = null;
+    private ArrayList<Integer> largeIntsList = null;
+
+    // Unequal sized lists
+    private static final int SMALL_LIST_SIZE = 20;
```

Comment:
If you expect the sizes to be different, you should assert that fact somewhere.

Reply:
Its asserted in two tests: `testLeftIteratorLargerThanRight` and `testRightIteratorLargerThanLeft`